### ISSUE TITLE
Project repo outside current course org

### ISSUE THREAD

**Issue by pconrad** [brownfield-team/anacapa-github-linker#341]
# User Story

As an admin/researcher, I can do analysis on repos that are outside the organization associate with the course in the GitHub linker, so that I can get research results for 
* Brownfield projects that move from courseOrg to courseOrg over time
* Brownfield projects that live outside the courseOrg for other reasons (e.g. capstone projects from industry partners, pre-existing open source project, etc.)

# Discussion

The current product allows us (once the Summer 2020 branch is merged), to extract various kinds of data from repos, e.g. commits, issues, issue activity, and in future iterations, information about PRs.

However, this can only be done on repos that live "inside" the GitHub organization associated with the course.  For various reasons, sometimes the project that students work on are in repos outside the organization.

# Acceptance Criteria

- [ ] There will be a way for instructors/admins to add project repos, specifying both the owner and the repo name (i.e. a new option on the Repo Menu, perhaps?  On the project repos page, a "New" button... this may already exist.)
- [ ] For all features that allow data retrieval on project repos (e.g. CSV of commits, CSV of issues, etc.), the code that gather that information will need to be modified so that it iterates through repos within the organizaiton AND those outside the organization.
- [ ] 

# Challenges:

* Setting up database tables, or modifying current ones, to keep track of such repos (right now, the organization/owner of every repo is "implied"--it's just always assumed in the code to be the current course org.

* Might need to split this into many smaller issue for modifying the code of each of the jobs and/or pages that get information (CSV download of issues, CSV download of commites, etc.) to modify each individually to provide for project repos outside the org.   This does NOT need to be a part of this particular story... but I would suggest at least doing a SPIKE on ONE of those before merging this story, just to make sure that whatever we put in place sets us up to make that feasible.

# Possible Future Issues, but out of scope for now.

* Permission: it is assumed that you always have permission to the repos in the course org via the Machine user, (in practice, is the GitHub user phtcon (Phill Conrad's alternate GitHub account.)    In this case, for example, in a Capstone class, we might need a machine user "per GitHub Repo", since the 3rd parties might not want to share their repo with Phill Conrad, but might be ok sharing it with Chris Hundhausen, for example.  Should we have a "machine users table" rather than a single machine user global to the entire app?

* It would be ok in the first cut at this, to separate out the "separate machine user piece" as a separate future story, since for Conrad's use case, it's a non-issue.. the existing global machine user already has permission to all of the repos under consideration.   The separate machine user is more of an issue for capstone like situations, such as might apply for Chris.

**Comment by pconrad** [comment]
As a separate PR, we may need to adjust the GraphQL queries to get information for the external project repos...
